contracts/lockdrop/src/contract.rs

```diff
@@ -1164,15 +1162,16 @@ pub fn update_pool_on_dual_rewards_claim(
     )?;
 
     let base_reward_received;
-    // Increment claimed Astro rewards per LP share
+    // Increment claimed rewards per LP share
     pool_info.generator_ntrn_per_share += {
-        let res: BalanceResponse = deps.querier.query_wasm_smart(
-            rwi.base_reward_token.to_string(),
-            &Cw20QueryMsg::Balance {
-                address: env.contract.address.to_string(),
-            },
-        )?;
-        base_reward_received = res.balance - prev_ntrn_balance;
+        let reward_token_balance = deps
+            .querier
+            .query_balance(
+                env.contract.address.clone(),
+                rwi.base_reward_token.to_string(),
+            )?
+            .amount;
+        base_reward_received = reward_token_balance - prev_ntrn_balance;
```

Comment:
While we are here, let's use `checked_sub` to feel :crab: extra safe :crab:?

Reply:
contract: becomes extra safe after single minus-to-checked_sub replacement
also contract: still has dozens of unsafe sub and sum operations

Reply:
@foxpy @swelf19 @oopcode you want it — you got it ;) please review